bundles/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/internal/service/TivoStatusProvider.java

```diff
@@ -0,0 +1,478 @@
+                        tivoConfigData.getCfgIdentifier());
+                StreamReader streamReader = this.streamReader;
+                if (streamReader != null && streamReader.isAlive()) {
+                    return true;
+                }
+            } else {
+                logger.debug(" connTivoConnect '{}' - Socket creation failed.", tivoConfigData.getCfgIdentifier());
+                TiVoHandler tivoHandler = this.tivoHandler;
+                if (tivoHandler != null) {
+                    tivoHandler.setStatusOffline();
+                }
+            }
+            // Sleep and retry
+            doNappTime();
+        }
+        return false;
+    }
+
+    /**
+     * {@link connTivoDisconnect} conditionally closes the Socket connection. When 'keep connection open' or 'channel
+     * scanning' is true, the disconnection process is ignored. Disconnect can be forced by setting forceDisconnect to
+     * true.
+     *
+     * @param forceDisconnect true = forces a disconnection , false = disconnects in specific situations
+     * @throws InterruptedException
+     */
+    public void connTivoDisconnect(boolean forceDisconnect) throws InterruptedException {
+        if (forceDisconnect) {
+            connSocketDisconnect();
+        } else {
+            if (!tivoConfigData.isKeepConnActive()) {
+                doNappTime();
+                connSocketDisconnect();
+            }
+        }
+    }
+
+    /**
+     * {@link connTivoReconnect} disconnect and reconnect the socket connection to the TiVo.
+     *
+     * @return boolean true = connection succeeded, false = connection failed
+     * @throws InterruptedException
+     */
+    public boolean connTivoReconnect() throws InterruptedException {
+        connSocketDisconnect();
+        doNappTime();
+        return connTivoConnect();
+    }
+
+    /**
+     * {@link connSocketDisconnect} cleanly closes the socket connection and dependent objects
+     *
+     */
+    private void connSocketDisconnect() {
+        TiVoHandler tivoHandler = this.tivoHandler;
+        StreamReader streamReader = this.streamReader;
+        PrintStream streamWriter = this.streamWriter;
+        Socket tivoSocket = this.tivoSocket;
+
+        logger.debug(" connTivoSocket '{}' - requested to disconnect/cleanup connection objects",
+                tivoConfigData.getCfgIdentifier());
+
+        // if isCfgKeepConnOpen = false, don't set status to OFFLINE since the socket is closed after each command
+        if (tivoHandler != null && tivoConfigData.isKeepConnActive()) {
+            tivoHandler.setStatusOffline();
+        }
+
+        try {
+            if (streamReader != null) {
+                streamReader.interrupt();
+                try {
+                    streamReader.join(TIMEOUT_SEC);
+                } catch (InterruptedException e) {
+                    logger.debug("Error joining streamReader: {}", e.getMessage());
+                }
```

Comment:
An interrupt isn't an error, it is another process requesting this thread to stop. You shouldn't silently consume an InterruptException either, at the very least you should have some way to notify the caller that an interrupt occured. Usually the best way to to rethrow the exception. But in this case you should wait to rethrow it until you have attempted to close the other other stuff first. With that in mind the best option would probably be to perform your `streamReader.join` at the end and then just add the InterruptException to the method signature.

Reply:
Done